src/System.Private.Xml.Linq/tests/TreeManipulation/SaveWithFileName.cs

```diff
@@ -0,0 +1,249 @@
+            using (System.IO.TextReader tr = new System.IO.StreamReader(filename)) {
+                result = StripOfXmlDeclaration(tr.ReadToEnd());
+            }
+            System.IO.File.Delete(filename);
+            return result;
+        }
+
+        [Fact]
+        public void XDocumentSaveToFile() {
+            SerializeWithSaveOptions(SerializeXDocumentSaveToFile, false, false, true);
+        }
+
+        [Fact]
+        public void XDocumentSave_NullParameter() {
+            Assert.Throws<ArgumentNullException>(() => new XDocument().Save((string)null));
+            Assert.Throws<ArgumentNullException>(() => new XDocument().Save((string)null, SaveOptions.DisableFormatting));
+            Assert.Throws<ArgumentNullException>(() => new XDocument().Save((string)null, SaveOptions.None));
+        }
+
+        private static string SerializeXElementSaveToFile(XNode node) {
+            string filename = System.IO.Path.GetTempFileName();
+            ((XElement)node).Save(filename);
+            string result;
+            using (System.IO.TextReader tr = new System.IO.StreamReader(filename)) {
+                result = StripOfXmlDeclaration(tr.ReadToEnd());
+            }
+            System.IO.File.Delete(filename);
+            return result;
+        }
+
+        [Fact]
+        public void XElementSaveToFile() {
+            SerializeWithSaveOptions(SerializeXElementSaveToFile, false, true, false);
+        }
+
+        [Fact]
+        public void XElementSave_NullParameter() {
+            Assert.Throws<ArgumentNullException>(() => new XElement("e").Save((string)null));
+            Assert.Throws<ArgumentNullException>(() => new XElement("e").Save((string)null, SaveOptions.DisableFormatting));
+            Assert.Throws<ArgumentNullException>(() => new XElement("e").Save((string)null, SaveOptions.None));
+        }
+
+        [Fact]
+        public void XStreamingElementSave_NullParameter() {
+            Assert.Throws<ArgumentNullException>(() => new XStreamingElement("e").Save((string)null));
+            Assert.Throws<ArgumentNullException>(() => new XStreamingElement("e").Save((string)null, SaveOptions.DisableFormatting));
+            Assert.Throws<ArgumentNullException>(() => new XStreamingElement("e").Save((string)null, SaveOptions.None));
+        }
+
+        //
+        // helpers
+        //
+        private delegate string SerializeNode(XNode node);
+
+        private static void SerializeWithSaveOptions(SerializeNode serialize, bool testXText, bool testXElement, bool testXDocument) {
+            // Test both options at once as they don't relly collide
+            SaveOptions so = SaveOptions.DisableFormatting | SaveOptions.OmitDuplicateNamespaces;
+
+            XElement root = XElement.Parse("<root xmlns:a='uri'><child xmlns:a='uri'><baby xmlns:a='uri'>text</baby></child></root>");
+            XElement child = root.Element("child"); XElement baby = child.Element("baby"); XNode text = baby.FirstNode;
```

Comment:
wierd. I would change it

---
In reply to: [85810573](https://github.com/dotnet/corefx/pull/13143#discussion_r85810573) [](ancestors = 85810573)